reference/zrc6.scilla

```diff
@@ -0,0 +1,963 @@
+    _amount: Uint128 0;
+    royalty_fee_bps: fee_bps
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Sets `uri` as the base URI. *)
+(* @Requirements: *)
+(* - `_sender` should be the contract owner. Otherwise, it should throw `NotContractOwnerError` *)
+transition SetBaseURI(uri: String)
+  RequireContractOwner;
+  base_uri := uri;
+
+  e = { 
+    _eventname: "SetBaseURI";
+    base_uri: uri
+  };
+  event e;
+  msg_to_sender = { 
+    _tag: "ZRC6_SetBaseURICallback"; 
+    _recipient: _sender;
+    _amount: Uint128 0;
+    base_uri: uri
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs  
+end
+
+(* Mints a token and transfers it to `to`. *)
+(* @param: to - Address of the token recipient   *)
+(* @Requirements: *)
+(* - The contract should not be paused. Otherwise, it should throw `PausedError` *)
+(* - `_sender` should be a minter. Otherwise, it should throw `NotMinterError` *)
+transition Mint(to: ByStr20)
+  MintToken to;
+  token_id <- token_id_count;
+  msg_to_recipient = { 
+    _tag: "ZRC6_RecipientAcceptMint";
+    _recipient: to;
+    _amount: Uint128 0
+  };
+  msg_to_sender = { 
+    _tag: "ZRC6_MintCallback";
+    _recipient: _sender;
+    _amount: Uint128 0;
+    to: to;
+    token_id: token_id
+  };
+  msgs = two_msgs msg_to_recipient msg_to_sender;
+  send msgs
+end
+
+(* Mints tokens and transfers them to `to_list`. *)
+(* @param: to_list - List of addresses of the token recipient *)
+(* @Requirements: *)
+(* - The contract should not be paused. Otherwise, it should throw `PausedError` *)
+(* - `_sender` should be a minter. Otherwise, it should throw `NotMinterError` *)
+transition BatchMint(to_list: List ByStr20)
+  forall to_list MintToken;
+  msg_to_sender = { 
+    _tag: "ZRC6_BatchMintCallback";
+    _recipient: _sender;
+    _amount: Uint128 0
+  };
+  msgs = one_msg msg_to_sender;
+  send msgs
+end
+
+(* Destroys `token_id`. *)
+(* @param: token_id - Unique ID of the NFT to be destroyed *)
+(* @Requirements: *)
+(* - The contract should not be paused. Otherwise, it should throw `PausedError` *)
+(* - `token_id` should exist. Otherwise, it should throw `TokenNotFoundError` *)
+(* - `_sender` should be a token owner or an operator. Otherwise, it should throw `NotOwnerOrOperatorError` *)
+transition Burn(token_id: Uint256)
+  RequireNotPaused;
+  (* Check if token exists *)
+  opt_token_owner <- token_owners[token_id];
+  match opt_token_owner with
+  | None =>
+    error = TokenNotFoundError;
+    Throw error
+  | Some token_owner =>
+    RequireOwnerOrOperator token_owner;
+    (* Destroy existing token *)
+    delete token_owners[token_id];
+    delete spenders[token_id];
+
+    (* subtract one from the balance *)
+    UpdateBalance sub_operation token_owner;
+    (* subtract one from the total supply *)
+    current_supply <- total_supply;
+    new_supply = builtin sub current_supply one;
+    total_supply := new_supply;
+    e = {
+      _eventname: "Burn";
+      burn_address: token_owner;
```

Comment:
can consider to rename to `burner`

`burn_address` typically refer to 0x0 or 0xdead

Reply:
I agree `burn_address` is not good. but `burner` is also not ideal IMO because operator can burn a token.
I think `token_owner` is OK.